### ISSUE TITLE
[BUG] KeyError: 'data' when calling method "hashtag_medias_recent(...)"

### ISSUE THREAD

**Issue by kqee** [subzeroid/instagrapi#1568]
**Describe the bug**
an attempt to call the method "Client().hashtag_medias_recent()" raised an exception, namely; KeyError.

**To Reproduce**
```py
from instagrapi import Client
client = Client()
client.hashtag_medias_recent('tree', amount=5)
```

**Traceback**
```
File ~\code\instagram\.venv\Lib\site-packages\instagrapi\mixins\hashtag.py:444, in HashtagMixin.hashtag_medias_recent(self, name, amount)
    428 """
    429 Get recent medias for a hashtag
    430
   (...)
    441     List of objects of Media
    442 """
    443 try:
--> 444     medias = self.hashtag_medias_recent_a1(name, amount)
    445 except ClientError:
    446     medias = self.hashtag_medias_recent_v1(name, amount)

File ~\code\instagram\.venv\Lib\site-packages\instagrapi\mixins\hashtag.py:407, in HashtagMixin.hashtag_medias_recent_a1(self, name, amount)
    391 def hashtag_medias_recent_a1(self, name: str, amount: int = 71) -> List[Media]:
    392     """
    393     Get recent medias for a hashtag by Public Web API
    394
   (...)
    405         List of objects of Media
    406     """
--> 407     return self.hashtag_medias_a1(name, amount, tab_key="recent")

File ~\code\instagram\.venv\Lib\site-packages\instagrapi\mixins\hashtag.py:222, in HashtagMixin.hashtag_medias_a1(self, name, amount, tab_key)
    202 def hashtag_medias_a1(
    203     self, name: str, amount: int = 27, tab_key: str = ""
    204 ) -> List[Media]:
    205     """
    206     Get medias for a hashtag by Public Web API
    207
   (...)
    220         List of objects of Media
    221     """
--> 222     medias, _ = self.hashtag_medias_a1_chunk(name, amount, tab_key)
    223     if amount:
    224         medias = medias[:amount]

File ~\code\instagram\.venv\Lib\site-packages\instagrapi\mixins\hashtag.py:179, in HashtagMixin.hashtag_medias_a1_chunk(self, name, max_amount, tab_key, end_cursor)
    176     self.inject_sessionid_to_public()
    177     data = self.public_a1_request(url, params=params)
--> 179 result = data["data"][tab_key]
    180 for section in result["sections"]:
    181     layout_content = section.get("layout_content") or {}

KeyError: 'data'
```
**Expected behavior**
'hashtag_medias_recent(...)' should return List[Media], not an error,

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 11,  10.0.22621 N/A Build 22621
 - Python version: Python 3.11.4
 - instagrapi version: 2.0.0

**Additional context**
same error also occurs with 'hashtag_medias_top(...)'

**Comment by Kweiza** [comment]
I have same issue
```
OS: macOS 13.5.1
Python version: Python 3.11.5
instagrapi version: 2.0.0
```

**Comment by kqee** [comment]
this must occur due to a change in [https://github.com/subzeroid/instagrapi/blob/198e09604e20c5559509d2f18d8ea04da9f627b0/instagrapi/mixins/hashtag.py#L179C11-L179C11](this line).

**Comment by kqee** [comment]
this must occur due to a change in [this line](https://github.com/subzeroid/instagrapi/blob/198e09604e20c5559509d2f18d8ea04da9f627b0/instagrapi/mixins/hashtag.py#L179C11-L179C11). I'll try fixing it.

**Comment by kqee** [comment]
## resolved?
I tried to reproduce this error today's evening, I was met with a different error; `LoginRequired: login_required`, to resolve, I logged in with the method `Client().login()`.

everything works fine now, for some reason. There must be some API changes going on,

**Comment by bazuker** [comment]
Looks like the JSON output has changed.

The JSON root key is no longer "data" and does not contain "top" or "recent". The root object is now "hashtag". 

Here is the JSON I got from `cl.hashtag_medias_recent('downhill', amount=2)`
[hashtag.json.zip](https://github.com/subzeroid/instagrapi/files/12554862/hashtag.json.zip)

**Comment by bazuker** [comment]
@kqee looks like `hashtag_medias_recent_v1(tag, amount)` can be used meanwhile.

**Comment by kqee** [comment]
resolved

**Comment by jarrodnorwell** [comment]
Fixed this in #1664
